src/print.cc

```diff
@@ -269,6 +267,7 @@ void send_recv_tile(
     if (tile_rank != 0) {
         if (A.tileIsLocal(i, j)) {
             try {
+                A.tileGetForReading( i, j, LayoutConvert::None );
```

Comment:
This is fix for hang if tile was on device. I'm not entirely sure why it would hang instead of throwing an exception.